### PR TITLE
Rustdoc: update Source Serif Pro and replace Heuristica italic

### LINKED ISSUE #57363: Rustdoc uses Heuristica as italic substitute for Source Serif Pro
[Source Serif Pro](https://github.com/adobe-fonts/source-serif-pro) has a matching italic font since August 2018, although that is not available in for example [fonts.google.com](https://fonts.google.com/specimen/Source+Serif+Pro). I think Heuristica should be replaced by the italic font of Source Serif Pro.

### PR DESCRIPTION
**Pull request by tspiteri**
When Source Serif Pro was used to replace Heuristica in #15530, the italic variant was not ready yet, but now it is. This PR updates the Source Serif Pro font files to the [latest release](https://github.com/adobe-fonts/source-serif-pro/releases/tag/2.007R-ro%2F1.007R-it) which includes an italic variant, and replaces Heuristica italic with Source Serif Pro italic.

Fixes #57363.

### CODE REVIEW

File: src/librustdoc/html/render.rs [hunk 1]
```diff
@@ -893,14 +893,12 @@ themePicker.onblur = handleThemeButtonsBlur;
           static_files::fira_sans::MEDIUM)?;
     write(cx.dst.join("FiraSans-LICENSE.txt"),
           static_files::fira_sans::LICENSE)?;
-    write(cx.dst.join("Heuristica-Italic.woff"),
-          static_files::heuristica::ITALIC)?;
-    write(cx.dst.join("Heuristica-LICENSE.txt"),
-          static_files::heuristica::LICENSE)?;
-    write(cx.dst.join("SourceSerifPro-Regular.woff"),
+    write(cx.dst.join("SourceSerifPro-Regular.ttf.woff"),
```
Comment (ishitatsuyuki) [inline src/librustdoc/html/render.rs]: This file name is... too weird.

**Comment by QuietMisdreavus** [comment]
cc @rust-lang/rustdoc, this PR changes our font set. We'll need to make sure to update docs.rs if/when this makes it to a nightly.

I'd like to merge this, since it makes our font set more consistent, but i feel like this is a foundational enough change that i would like to make sure the rest of the team is aware of it.

**Comment by GuillaumeGomez** [comment]
That is **really** awesome! 👍

**Comment by GuillaumeGomez** [comment]
@QuietMisdreavus I think this is ready and no one add complaints, so I'll r+.

Thanks @tspiteri!

@bors: r+ rollup

**Comment by pietroalbini** [comment]
@bors retry
AppVeyor... what's wrong with you today?

**Pull request merged**